Pull Timestamp container into Psammead.

Bopchy opened the pull request:
Resolves https://github.com/bbc/simorgh/issues/1570

**Overall change:** _Pull TimestampContainer form Simorgh into Psammead._

**Code changes:**

- _Create containers/psammead-timestamp-container and update simorgh specific imports._
- _Make script a TimestampContainer prop._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

Commit:
Pull Timestamp container into Psammead.
- Create containers/psammead-timestamp-container and update simorgh specific imports.
- Make script a TimestampContainer prop.

Commit:
Update CHANGELOG.md with PR number and link.

Commit:
Update tests. Update snapshots.

Commit:
Add tests to increase coverage.

Commit:
Fix eslint issues. Update stories to use knobs.

Commit:
Add content to README.

Bopchy commented on packages/containers/psammead-timestamp-container/package.json:
```diff
@@ -0,0 +1,26 @@
+  },
+  "license": "Apache-2.0",
+  "keywords": [
+    "bbc"
```
I am not sure what other keywords should be added here

sareh replied:
You can include values such as `timestamp` `datetime` `relative time` `absolute time` `moment-timezone`

Commit:
Update README.

Commit:
Merge branch 'latest' of github.com:bbc/psammead into 1570-move-timestamp-container-to-psammead

twinlensreflex committed:
Merge branch 'latest' into 1570-move-timestamp-container-to-psammead

twinlensreflex commented on packages/containers/psammead-timestamp-container/src/relativeTimestamp.js:
```diff
@@ -0,0 +1,36 @@
+  let timeDiff;
+  let unit;
+
+  if (timeDiffMins >= 1440) {
```
could we store 1440 in a variable like `twentyFourHours` or something similar so it's easy to see what the significance of 1440 is without googling?

twinlensreflex commented on packages/containers/psammead-timestamp-container/src/relativeTimestamp.js:
```diff
@@ -0,0 +1,36 @@
+  if (timeDiffMins >= 1440) {
+    timeDiff = parseInt(timeDiffMins / 1440, 10);
+    unit = timeDiff > 1 ? 'days' : 'day';
+  } else if (timeDiffMins >= 60) {
```
same here, maybe store 60 as a variable of `oneHour` ?

twinlensreflex commented on packages/containers/psammead-timestamp-container/src/relativeTimestamp.js:
```diff
@@ -0,0 +1,36 @@
+    unit = timeDiff > 1 ? 'days' : 'day';
+  } else if (timeDiffMins >= 60) {
+    timeDiff = parseInt(timeDiffMins / 60, 10);
+    unit = timeDiff > 1 ? 'hours' : 'hour';
```
since the content of these `if/else` statements has a bit of duplication we could create a function like this:
```js
const pluralise = (timeDiffMins, timeUnit, singular) => {
    timeDiff = parseInt(timeDiffMins / timeUnit, 10);
    unit = timeDiff > 1 ? `${singular}s` : singular;
}
```

then you could call it
`pluralise(timeDiffMins, oneHour, 'hour')`
in each of the above scenarios. I think this would reduce the cognitive complexity.

twinlensreflex commented on packages/containers/psammead-timestamp-container/src/relativeTimestamp.js:
```diff
@@ -0,0 +1,36 @@
+    unit = timeDiff > 1 ? 'hours' : 'hour';
+  } else {
+    timeDiff = timeDiffMins <= 1 ? 1 : timeDiffMins;
+    unit = timeDiff === 1 ? 'minute' : 'minutes';
```
these lines are also increasing Code Climate's cognitive complexity but I like your approach so I think it's okay 👍 (that said, if you could think of a way to put this logic in the `pluralise` function I think code climate would be happy.. it doesn't like nested `if/else`s)

twinlensreflex reviewed (commented):
it looks good to me so far 😄 I've made a few comments on `packages/containers/psammead-timestamp-container/src/relativeTimestamp.js` to make it easier to understand at first glance

sareh commented on packages/containers/psammead-timestamp-container/README.md:
````diff
@@ -0,0 +1,53 @@
+    script={latin}
+  />
+);
+```
````
We should have a section here for  `## Accessibility notes` & another for the ` ## Roadmap `

All of the required sections in the Readme are listed here in the Contributing.md file. https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md

The Accessibility section can include notes similar to that for the Psammead Timestamp, since it is using this as a dependency. https://github.com/bbc/psammead/blob/latest/packages/components/psammead-timestamp/README.md#accessibility-notes

The Roadmap section can include a note that this package currently only supports Relative strings for English and the plan is to include logic to render relative time in multiple languages, for use on the World Service sites.

sareh commented on packages/containers/psammead-timestamp-container/package.json:
```diff
@@ -0,0 +1,26 @@
+  "name": "@bbc/psammead-timestamp-container",
+  "version": "1.0.0",
+  "main": "dist/index.js",
+  "description": "Generic Timestamp react container",
```
```suggestion
  "description": "React Timestamp container for relative and absolute times, using moment-timezone",
```

sareh commented on packages/containers/psammead-timestamp-container/README.md:
```diff
@@ -0,0 +1,53 @@
+# psammead-timestamp-container - [![Known Vulnerabilities](https://snyk.io/test/github/bbc/psammead/badge.svg?targetFile=packages%2Fcontainers%2Fpsammead-timestamp-container%2Fpackage.json)](https://snyk.io/test/github/bbc/psammead?targetFile=packages%2Fcontainers%2Fpsammead-timestamp-container%2Fpackage.json) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-timestamp-container.svg)](ht ...
+
+## Description
+`psammead-timestamp-container` is a container with generic time functions, that returns a single time element.
```
This is the main introduction to this package, so we should explain what functionality it has. 
It has relative and absolute times with timezone support, using `moment-timezone`. 
Relative times are currently hard-coded to English (e.g. `3 minutes ago`).

sareh reviewed (changes requested):
@Bopchy Looks good so far. Just some comments regarding the documentation.

Commit:
Refactor to use pluralise(), removing repetition in constructTimestamp().

Commit:
Update package description.

Co-Authored-By: Sareh <sareh.heidari@bbc.co.uk>

Commit:
Update README.
- Add Accessibility and Raodmap section.
- Update Description.
- Update keywords to package.json.

Commit:
Merge branch '1570-move-timestamp-container-to-psammead' of github.com:bbc/psammead into 1570-move-timestamp-container-to-psammead

Commit:
Merge branch 'latest' of github.com:bbc/psammead into 1570-move-timestamp-container-to-psammead

Commit:
Update README

oluoluoxenfree commented on packages/containers/psammead-timestamp-container/README.md:
```diff
@@ -0,0 +1,62 @@
+
+## Accessibility notes
+
+The usage of the semantic `time` element within psammead-timestamp component can result in strange behaviours when using assistive technology such as VoiceOver for iOS. Specifically, it results in a "text splitting" behaviour, where any text in the prefix is read out, but the value inside the semantic `time` element require an additional swipe before being read out.
```
What do you mean by a prefix here? Like the sentence before or part of the timestamp?

Bopchy replied:
Here it means the sentence before - because it refers to the psammead-timestamp component that is being pulled in

sareh reviewed (approved):
👍 @Bopchy Thanks for the updates. Looks great!

jamesbrumpton committed:
Merge branch 'latest' into 1570-move-timestamp-container-to-psammead

Commit:
Merge branch 'latest' into 1570-move-timestamp-container-to-psammead
